community/detectors/gitlab_cve_2021_22205/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202122205/Cve202122205VulnDetector.java

```diff
@@ -0,0 +1,269 @@
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020200A0D0A2D2D2D2D2D"
+          + "2D5765624B6974466F726D426F756E64617279494D76336D7852673539546B465358352D2D0D0A";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve202122205VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown");
+  }
+
+  private static String buildTargetUserSignUrl(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(USER_SIGN_PATH);
+    return targetUrlBuilder.toString();
+  }
+
+  private static String buildTargetVulnerabilityUrl(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(VUL_PATH);
+    return targetUrlBuilder.toString();
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("CVE-2921-22205 starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve202122205VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private String clearCookie(String cookie) {
+    return cookie.split("path=/;")[0];
+  }
+
+  private Map<String, String> getCsrfTokenAndCookie(NetworkService networkService) {
```

Comment:
This returns a hashmap with two constant keys. Why not just return a class with two string fields (cookie and csrf token)?